locale.c

```diff
@@ -2578,10 +2578,12 @@ S_wrap_wsetlocale(pTHX_ const int category, const char *locale)
         return NULL;
     }
 
-    const char * result = Win_wstring_to_utf8_string(wresult);
-    SAVEFREEPV(result); /* is there something better we can do here? */
-
-    return result;
+    const char * result = (const char *) Win_wstring_to_utf8_string(wresult);
+    const char * retval = save_to_buffer(result,
+                                         &PL_win32setlocale_buf,
+                                         &PL_win32setlocale_bufsize);
+    Safefree(result);
```

Comment:
Wouldn't it be better to modify Win_wstring_to_byte_string() to accept the buf/bufsize pointers avoid the extra allocation/copy/free?

You'd need to extract the allocation part of save_to_buffer() out to a new function that Win_wstring_to_byte_string() would call.

Reply:
The code failed anyway.  This PR is now just a documentation change that things fail